varada_trino_manager/query_json_jstack.py

```diff
@@ -0,0 +1,85 @@
+from .connections import Trino, PrestoRest
+from .configuration import Connection
+from .utils import logger
+from click import exceptions
+from .constants import Paths
+from json import load, dump
+from pathlib import Path
+from .remote import parallel_ssh_execute, parallel_download
+from time import sleep
+from threading import Thread, Event
+
+
+def run_query(query: str, client: Trino) -> dict:
+    _, stats = client.execute(query=query)
+    logger.info(f'Query: {query} QueryId: {stats["queryId"]} '
+                f'Query execution time: {round(stats["elapsedTimeMillis"]*0.001, 3)} Seconds')
+    return {"queryId": stats["queryId"], "elapsedTime": round(stats["elapsedTimeMillis"]*0.001, 3)}
+
+
+def collect_jstack(wait: int, keep_running: Event):
+    collect_commands = [
+        "echo '#######################################    NEW JSTACK CAPTURE   #######################################' | sudo tee -a /tmp/jstacks/jstack.txt",
+        "sudo jps | awk '/TrinoServer/ {print $1}' | sudo tee -a /tmp/jstacks/server.pid",
+        "sudo jstack $(sudo cat /tmp/jstacks/server.pid) | sudo tee -a /tmp/jstacks/jstack.txt || true",
+    ]
+    while keep_running.is_set():
+        parallel_ssh_execute(command="\n".join(collect_commands))
+        sleep(wait)
+
+
+def run(user: str, con: Connection, jsonpath: Path, query: str, jstack_wait: int, dest_dir: str):
+    try:
+        with open(jsonpath) as fd:
+            queries = load(fd)
+    except Exception:
+        logger.exception(f'Failed reading {jsonpath}')
+        raise exceptions.Exit(code=1)
+
+    if query not in queries:
+        logger.error(f'Query {query} is not in {queries.keys()}')
+        raise exceptions.Exit(code=1)
+
+    dir_commands = [
+        "sudo rm -rf /tmp/jstacks",
+        "sudo rm -rf /tmp/jstacks.tar.gz",
+        "sudo mkdir /tmp/jstacks",
+    ]
+    parallel_ssh_execute(command="\n".join(dir_commands))
+
+    with Trino(con=con, username=user) as trino_client:
+        """
```

Comment:
this is not the place for doc string, if you do want to use it put it after the function declaration